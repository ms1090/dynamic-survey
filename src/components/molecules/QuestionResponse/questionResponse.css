.question-response {
  margin-bottom: 20px;
}
.question-response .question-container {
  font-weight: bold;
  margin: 10px 0;
}
.question-response .answer-container {
  color: #8c8c8a;
}
.question-response .question-container,
.question-response .answer-container strong {
  color: #285175;
}
.question-response .question-container span:last-child,
.question-response .answer-container span {
  margin-left: 5px;
}