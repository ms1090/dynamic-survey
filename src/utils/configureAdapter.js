import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// configure adapter to use enzyme in unit test cases
configure({ adapter: new Adapter() });
