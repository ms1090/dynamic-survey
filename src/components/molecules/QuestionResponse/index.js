import React from 'react';
import PropTypes from 'prop-types';
import './questionResponse.css';

/**
 * Display question and user answer
 * @param {Object} question Question details object
 * @param {number} index Question index in list
 * @return {React.Element} Return question and answer UI
 */
const QuestionResponse = ({ question, index }) => (
  <article className="question-response">
    <div className="question-container">
      <span>{`Q ${index} : `}</span>
      <span>{question.text}</span>
    </div>
    <div className="answer-container">
      <strong>A : </strong>
      <span>{question.answer || 'No answer'}</span>
    </div>
  </article>
);
// default values for component props
QuestionResponse.defaultProps = {
  question: {},
  index: 1,
};
// prop types checking
QuestionResponse.propTypes = {
  question: PropTypes.shape({
    id: PropTypes.string,
    options: PropTypes.arrayOf(PropTypes.string),
    text: PropTypes.string,
    answer: PropTypes.string,
  }),
  index: PropTypes.number,
};
export default QuestionResponse;
