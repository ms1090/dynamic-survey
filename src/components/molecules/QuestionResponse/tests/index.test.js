import React from 'react';
import { shallow } from 'enzyme';
import QuestionResponse from '..';
import toJson from 'enzyme-to-json';
import 'utils/configureAdapter';

describe('QuestionResponse Component', () => {
  test('should render correctly', () => {
    const question = { text: 'test question?' };
    const wrapper = shallow(<QuestionResponse
      question={question}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
