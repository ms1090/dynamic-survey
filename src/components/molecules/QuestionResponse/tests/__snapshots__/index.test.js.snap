// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionResponse Component should render correctly 1`] = `
<article
  className="question-response"
>
  <div
    className="question-container"
  >
    <span>
      Q 1 : 
    </span>
    <span>
      test question?
    </span>
  </div>
  <div
    className="answer-container"
  >
    <strong>
      A : 
    </strong>
    <span>
      No answer
    </span>
  </div>
</article>
`;
